extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/TaskRunner.java

```diff
@@ -132,7 +156,102 @@ public Map<String, String> configs() {
 
         @Override
         public OffsetStorageReader offsetStorageReader() {
-            return new SourceOffsetStorageReader(partitionsToOffset);
+            return new SourceOffsetStorageReader(state.partitionsToOffset);
+        }
+    }
+
+    public void commitRecord(SourceRecord rec) {
+        state.commitRecord(rec);
+        try {
+            if (running) {
+                task.commitRecord(rec, null);
+            }
+        } catch (InterruptedException ie) {
+            LOGGER.warning("Interrupted while committing record");
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public State createSnapshot() {
+        State snapshot = new State();
+        snapshot.load(state);
+        return snapshot;
+    }
+
+    public void restoreSnapshot(State state) {
+        this.state.load(state);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String toString() {
+        return "TaskRunner{" +
+                "name='" + name + '\'' +
+                '}';
+    }
+
+    @FunctionalInterface
+    interface SourceTaskFactory {
+        SourceTask create();
+    }
+
+    static class State implements Serializable {
+
+        /**
+         * Key represents the partition which the record originated from. Value
+         * represents the offset within that partition. Kafka Connect represents
+         * the partition and offset as arbitrary values so that is why it is
+         * stored as map.
+         * See {@link SourceRecord} for more information regarding the format.
+         */
+        private final Map<Map<String, ?>, Map<String, ?>> partitionsToOffset;
+
+        State() {
+            this(new ConcurrentHashMap<>());
```

Comment:
We can assign it directly to the field here.

Reply:
fixed https://github.com/hazelcast/hazelcast/pull/23831/commits/5839ab6a3b00631f55381d5fbbb2e305dea5eb13